tests/test_utils.py

```diff
@@ -0,0 +1,175 @@
+            self.mocked_logger
+        )
+
+        utils._prepare_plugins_for_installation.assert_called_once_with(
+            self.workdir,
+            [self.mock_plugin_name],
+        )
+        utils._log_unused_plugins.assert_called_once_with(
+            self.mocked_logger,
+            self.workdir,
+            [self.mock_plugin_name],
+        )
+        self.mocked_plugin_manager.validate_plugin.assert_called_once_with(
+            os.path.join(self.workdir, self.mock_plugin_name)
+        )
+        self.mocked_plugin_manager.install.assert_called_once_with(
+            os.path.join(self.workdir, self.mock_plugin_name)
+        )
+
+        assert len(self.mocked_logger.method_calls) == 2
+
+    def test_install_existing_plugin(self, mocker):
+        mocker.patch('aria_plugin.utils._prepare_plugins_for_installation',
+                     return_value=[self.mock_plugin_name])
+
+        def existing_plugin_error(*args, **kwargs):
+            raise aria_exception.PluginAlreadyExistsError()
+
+        self.mocked_plugin_manager.install = existing_plugin_error
+
+        with pytest.raises(exceptions.PluginsAlreadyExistException):
+            utils.install_plugins(
+                self.workdir,
+                [self.mock_plugin_name],
+                self.mocked_plugin_manager,
+                self.mocked_logger
+            )
+
+    def test_non_existing_plugins_dir(self, mocker):
+        mocker.patch('aria_plugin.utils._prepare_plugins_for_installation',
+                     return_value=[self.mock_plugin_name])
+        with pytest.raises(exceptions.MissingPluginsException):
+            utils.install_plugins(
+                'non_existing_dir',
+                '[plugin_to_install',
+                'plugin_manager',
+            )
+
+    def test_prepare_plugin_for_installation(self, mocker):
+        mocker.patch('os.listdir', return_value=['plugin1'])
+
+        installed_plugins = utils._prepare_plugins_for_installation(
+            'mock_dir',
+            ['plugin1']
+        )
+        assert installed_plugins == set(['plugin1'])
+
+        with pytest.raises(exceptions.MissingPluginsException):
+            utils._prepare_plugins_for_installation('mock_dir', ['plugin2'])
+
+
+def test_extract_csar(mocker):
+    mocker.patch('aria.cli.csar.read')
+    utils.extract_csar('mock_source', 'mock_logger')
+
+    _, kwargs = csar.read.call_args
+    assert kwargs['source'] == 'mock_source'
+    assert kwargs['logger'] == 'mock_logger'
+
+
+def test_generate_resource_path():
+    local_resource = os.path.join('dir', 'some_path')
+    assert local_resource == utils.generate_resource_path('some_path', 'dir')
+
+    url_resource = 'http://some_resource'
+    assert utils.generate_resource_path(url_resource, 'dir') == url_resource
+
+
+def test_cleanup_files(tmpdir):
+
+    def _touch_files(files):
+        for f in files:
+            f.write('content')
+
+    files = [tmpdir.join(f) for f in ('file1', 'file2')]
```

Comment:
maybe add comments over the following two subsections that explains what each does in a short sentence